src/cookbook/gradual-compositionapi-migration.md

```diff
@@ -0,0 +1,255 @@
+# Gradual Composition API Migration
+
+The [Composition API](/guide/composition-api-introduction.html) is an advanced feature introduced in Vue 3. It is purely additive, and the Options API is not legacy. You may find, however, that the Composition API can be a useful feature for more flexible large-scale architectures. So how would we go about introducing it in a larger system, gradually? What follows is merely a recommendation, there are many potential ways forward.
+
+Due to the flexibility of the Composition API, you may need to be more explicit about organization so that future maintainers can quickly understand intended purpose of pieces of the application.
+
+## Refactor your Mixins
+
+If you've made use of mixins for reusable logic across components, it's recommended that you start your refactor here. Mixins translate very directly.
+
+### Base Example
+
+Let's say we had this very small toggle mixin in `mixins/toggle.js`:
+
+```js
+const toggle = {
+  data() {
+    return {
+      isShowing: false
+    }
+  },
+  methods: {
+    toggleShow() {
+      this.isShowing = !this.isShowing
+    }
+  }
+}
+```
+
+And it was being used in a modal component in `components/modal.vue`:
+
+```html
+<template>
+  <div>
+    <h3>Let's trigger this here modal!</h3>
+
+    <button @click="toggleShow">
+      <span v-if="isShowing">Hide child</span>
+      <span v-else>Show child</span>
+    </button>
+
+    <div v-if="isShowing" class="modal">
+      <h2>Here I am!</h2>
+      <button @click="toggleShow">Close</button>
+    </div>
+  </div>
+</template>
+```
+
+```js
+<script>
+  import { toggle } from '@/mixins/toggle';
+
+  export default {
+    mixins: [toggle]
+  }
+</script>
+```
+
+We could refactor this by creating a `composables` folder and create `composables/useToggle.js` instead. We suggest using a directory named `composables` so that you can communicate that this is being used slightly differently from a component, it's reusable logic that you can consume.
```

Comment:
This is a larger thought that I've been thinking on for some time, using this line as a way to open discussion.

I know this is starting to become more regularly used, but I'm thinking that `features` is a better name than `composables` due to the fact that these files are not required to be used in the Composition API context. Because of how Vue 3 has been rewritten internally, it would actually be more flexible to allow users the mental model of combining `features` together outside the component context as well. So in other word, `composables` seems to imply that these things are Vue specific, when other than some Vue magic, they're generally not and probably should not be given that impression.

The closest analogy to this is how `utilities` is a common folder that anyone who understand English and some programming would catch on quickly. And similarly, `features` would lower the barrier of entry. 

And in the case there are functions that require to be used in a Vue context (i.e., utilizes into lifecycle methods), then we could recommend a prefix like `setupToggle` which would be very explicit as far to its usage sine that would correlate with the `setup` method where it needs to be used.

Similarly, still need to test this out, but I wonder if `use` as a prefix will ultimately tie the concept too closely to React Hooks when the Composition API is in fact simpler and quite different from a usage / implementation perspective. And since as a verb it's fairly generic, like we would never name the lodash utilities function file "useLodash" even though we do "use" it, perhaps we should be setting precedent for more declarative verbs. So like if we're creating a new instance of toggle, "createToggle" would be more intuitive from a naming perspective.

What do you all think? @NataliaTepluhina @sdras @phanan

Reply:
I completely agree because for me it's actually tougher to scan so easily- `composables` and `components` are close enough in spelling that it takes me one more second to understand which one is which. I like `features` way better myself.

Reply:
This is updated! @yyx990803 when you take a look at this, you might want to sound off as we're defining a new convention.

Reply:
Oh, for the `setup` vs `use`- I agree but didn't make that change yet because I think there are already so many examples out there that have `use` as a convention, including the VueUse site, that there would probably be bigger community impacts there.

Reply:
For those following along, we chatted with Evan, we decided that the ship has already sailed on `use` vs `setup` though Ben makes an excellent point. We also decided to stick to `composables` for now- I'll be reverting that.